Add file for GNU build on Hera

JulieSchramm opened the pull request:
## DESCRIPTION OF CHANGES: 
- Add file env/build_hera_gnu.env
- Replace deprecated add_stylesheet with add_css_file for documentation build on ReadTheDocs

## TESTS CONDUCTED: 
GNU build on Hera successful.  Modifications needed for regional workflow to run to completion.

|  GST                |  Hera         | Cheyenne      |
|---------------| ---------- | ------------- |
| Intel Build        | Pass          | Pass              |
| Intel Run          | Pass          | Pass              |
| GNU Build       | Pass          | Pass              |
| GNU Run         | Pass          | Pass              |


## DEPENDENCIES:
- See [PR#526](https://github.com/NOAA-EMC/regional_workflow/pull/526)

## DOCUMENTATION:
Modified documentation to build.  This will include information on setting COMPILER="gnu" in the config.sh file.

## ISSUE (optional): 
Fixes issue mentioned in #98

Commit:
Add env file for GNU build on Hera.

Commit:
Replace deprecated add_stylesheet with add_css_file to fix error:
AttributeError: 'Sphinx' object has no attribute 'add_stylesheet'

Commit:
Merge branch 'develop' into feature/hera_gnu_build

Commit:
Merge branch 'feature/hera_gnu_build' of https://github.com/JulieSchramm/ufs-srweather-app into feature/hera_gnu_build

Commit:
Merge branch 'develop' into feature/hera_gnu_build

Commit:
Merge branch 'ufs-community:develop' into feature/hera_gnu_build

Commit:
Merge branch 'feature/hera_gnu_build' of https://github.com/JulieSchramm/ufs-srweather-app into feature/hera_gnu_build

gsketefian commented:
Looks good!

JeffBeck-NOAA commented on env/build_hera_gnu.env:
```diff
@@ -0,0 +1,50 @@
+
+
+# For run in bash
+export CMAKE_C_COMPILER=mpicc
```
Can you make this say "for build and/or run in bash"?

Commit:
- Update hash to latest version of regional_workflow: 59e09bc
- Add setenv for csh build in env/build_hera_intel.env and env/build_hera_gnu.env
- Update versions of cmake, g2, g2tmpl, and add fms library in env/build_hera_gnu.env for updated ufs-weather-model
